include/picongpu/param/density.param

```diff
@@ -223,24 +223,35 @@ namespace picongpu
         /* definition of sphere profile with flanks */
         using SphereFlanks = SphereFlanksImpl<SphereFlanksParam>;
 
-
+        /** Density values taken from a file
+         *
+         * The file follows the openPMD standard.
+         * The density values are written as a dataset of type float_X.
+         * Each value defines density in the cell with the corresponding total coordinate minus the given offset.
+         * When the functor is instantiated, it will load the part matching the current domain position.
+         * If the file dataset is not large enough, it will be padded with the given defaultDensity value.
+         * Dimensionality of the file indexing must match the simulation dimensionality.
+         * Density values are in BASE_DENSITY_SI units.
+         */
         PMACC_STRUCT(
-            FromHDF5Param,
-            /* prefix of filename
-             * full file name: gas_0.h5
-             * filename = "gas"
-             * iteration = 0
-             */
-            (PMACC_C_STRING(filename, "gas"))
+            FromOpenPMDParam,
+            //! Name for the openPMD file
+            (PMACC_C_STRING(filename, "density.h5"))
 
-                (PMACC_C_STRING(datasetName, "fields/e_chargeDensity"))
+            //! Name of the dataset inside the file
+            (PMACC_C_STRING(datasetName, "density"))
 
-            /* simulation step*/
-            (PMACC_C_VALUE(uint32_t, iteration, 0))(
-                PMACC_C_VALUE(float_X, defaultDensity, 0.0))); /* struct FromHDF5Param */
+            //! Iteration inside the file (only file, not related to the current simulation time iteration)
+            (PMACC_C_VALUE(uint32_t, iteration, 0))
 
-        /* definition of cloud profile */
-        using FromHDF5 = FromHDF5Impl<FromHDF5Param>;
+            //! Offset in the file in cells: each value is density at (total cell index - offset)
+            (PMACC_C_VECTOR_DIM(int, simDim, offset, 0, 0, 0))
```

Comment:
Is this `offset` shifting the density defined in the file to the `right` or skipping data within the file?

Reply:
IMO if I use a negative offset my data will be loaded late in the simulation so I will add zeros in front of what I have defined in the file and if the offset is positive it is skipping/ignoring the first cells in the file.

Is this correct?

Reply:
It just shifts the indexing. So before I added this variable, the values in the file were total cell index. And now they are (total cell index - offset). You are right that the padding should be working for this case as well, I didn't think of that.

Reply:
Fixed now

Reply:
Are positive and negative shifts allowed? IMO the offset is not correctly handled in the implementation.

Reply:
@sbastrakov You missed this comment!

Reply:
Pushed the updated version that I forgot to push earlier.